src/Assignments/SimpleProblemPage.tsx

```diff
@@ -6,98 +6,305 @@ import ProblemIframe from './ProblemIframe';
                     return;
                 }
-                const res = await getQuestions({
-                    userId: 'me',
-                    courseTopicContentId: parseInt(params.topicId, 10)
-                });
-                const problems: Array<ProblemObject> = res.data.data.questions;
-                
-                const topic = res.data.data.topic;
-
-                if (topic.studentTopicOverride?.length > 0) {
-                    _.assign(topic, topic.studentTopicOverride[0]);
-                }
-                setTopic(topic);
-
-                if (!_.isEmpty(problems)) {
-                    const problemDictionary = _.chain(problems)
-                        .map(prob => {
-                            const override = prob.studentTopicQuestionOverride?.[0];
-                            if (!_.isNil(override)) {
-                                delete override.id;
-                                _.assign(prob, override);
-                            }
-                            return new ProblemObject(prob);
-                        })
-                        .keyBy('id')
-                        .value();
-                    setProblems(problemDictionary);
-                    setSelectedProblemId(problems[0].id);
-                }
 
+                await fetchProblems(parseInt(params.topicId, 10));
                 setLoading(false);
             } catch (e) {
                 setError(e.message);
                 console.error(e);
                 setLoading(false);
             }
         })();
-    }, [params.topicId]);
+    }, [params.topicId, versionId]);
+
+    const fetchProblems = async (topicId: number) => {
+        const res = await getQuestions({
+            userId: 'me',
+            courseTopicContentId: topicId
+        });
+        const problems: Array<ProblemObject> = res.data.data.questions;
+
+        const topic = res.data.data.topic;
+
+        if (topic.studentTopicOverride?.length > 0) {
+            _.assign(topic, topic.studentTopicOverride[0]);
+        }
+        setTopic(topic);
+
+        if (!_.isEmpty(problems)) {
+            const problemDictionary = _.chain(problems)
+                .map(prob => {
+                    const override = prob.studentTopicQuestionOverride?.[0];
+                    if (!_.isNil(override)) {
+                        delete override.id;
+                        _.assign(prob, override);
+                    }
+                    return new ProblemObject(prob);
+                })
+                .keyBy('id')
+                .value();
+            setProblems(problemDictionary);
+            setSelectedProblemId(_.sortBy(problems, ['problemNumber'])[0].id);
+            if (!_.isNil(topic.topicAssessmentInfo) &&
+                !_.isNil(topic.topicAssessmentInfo.studentTopicAssessmentInfo) &&
+                topic.topicAssessmentInfo.studentTopicAssessmentInfo.length > 0 // student has generated at least one version already
```

Comment:
If Typescript infers the null correctly, this would be cleaner.
```suggestion
            if (!_.isNil(topic.topicAssessmentInfo?.studentTopicAssessmentInfo) &&
                !_.isEmpty(topic.topicAssessmentInfo.studentTopicAssessmentInfo) // student has generated at least one version already
```

Reply:
this breaks other refs to topic.topicAssessmentInfo fields

Reply:
Yeah, if I'm understanding the problem then this is one of the things that annoys me about typescript
If you use nil chaining and a nil check it doesn't determine all types in the chain to be not nil
For example
```Typescript
        const  r: Q | undefined = {};
        if (
            !_.isNil(r) &&
            !_.isNil(r.a) &&
            !_.isNil(r.a.b) &&
            !_.isNil(r.a.b.c)
        ) {
            console.log(r.a.b.c.d);
        }
        if (!_.isNil(r?.a?.b?.c)) {
            console.log(r.a.b.c.d);
        }
```

The above two cases should be equivalent but they are not, the bottom one says that all paths can be undefined except c (it doesn't check d since it isn't being chained)

Reply:
If only I read @PanopticaRising's comment more closely 🤣 
> If Typescript infers the null correctly